src/marketmap.rs

```diff
@@ -103,15 +102,16 @@ impl<T: AccountDeserialize + Clone + Send + Sync + Market + 'static> MarketMap<T
             self.sync().await?;
         }
 
-        if !self.subscribed.get() {
-            self.subscription.try_borrow_mut()?.subscribe::<T>().await?;
-            self.subscribed.set(true);
+        if !self.subscribed.load(Ordering::Relaxed) {
```

Comment:
💭 an idea to remove the synchronized subscribe bool, can encode state via generics
```
struct Subscribed;

impl <T> MarketMap<Subscribed, T> {
  /// subscribed state methods
}

struct Unsubscribed;

impl<T> MarketMap<Unsubscribed, T> {
  /// unsubscribed state methods
}
```